### DISCUSSION TITLE
可以将亮度调节的显示风格与win11 22H2版本的原生风格统一一下

### DISCUSSION THREAD

**Discussion by bonjour-cell** [shuax/MouseInc.Settings#448] [Ideas]
![image](https://user-images.githubusercontent.com/28861466/192749519-1646ce28-ba7b-41a5-bce4-9be57af0a219.png)
![image](https://user-images.githubusercontent.com/28861466/192749555-de6f820a-37d6-4391-a2ed-9a07ac082eee.png)
上图是现在的风格，下图是新的风格

**Comment by SRH440** [comment]
是的 我也很希望能够统一一下。

**Comment by 6Tom** [comment]
1.大佬已经退网了

2.加入图形界面需要设计和做进度条，可能需要集成一个或多个png，这是个全新的工作

3.你有这个想法可能是出于音量调节吧，那是系统自带的，不是大佬做的，这是不一样的

4.不显示数值有啥可好的，我十分感谢很多开发者做了第三方软件可以显示数值，没有数值连个常用参考都没有

**Reply by bonjour-cell** [reply]
1.退网这件事在我提出这条建议之后，这件事我知道。
2&3. 更新22H2之前，音量条和亮度条均为图1的风格；更新22H2之后，音量调节才变为图2的样子。可以看出，作者此前似乎是调用了原生的音量接口（说法可能不准确）而没有调用原生的亮度接口。我不懂你说的重新设计是出于什么考虑。
4.现在MouseInc在调节音量时显示数值。

**Reply by bonjour-cell** [reply]
此外，我发的就是Ideas区。请问提出建议是有哪里不妥吗？

**Reply by 6Tom** [reply]
1. 我给你按123分点列，只是为了逻辑清晰，便于你阅读，并不是我语气强硬，这是个误会，先跟你道个歉，因为文字没发体现语气，我不考虑组织逻辑的话，读起来就会很困难

2. 我只是在帮你解释一些我恰好知道的事情，因为大佬退了，怕没人给你解释你不知道

3. 我看了你是个开发者啊，那么你也应该知道一些，就是不太成文的规则，就是，**在个人开发者的项目中提问要谨慎一些，不代表以ideas的形式提问，就可以天马行空了啊**，虽然软件不是开源的，但也是纯粹的用爱发电，也不接受捐助，所以提问应该和开源社区一样，首先要换位思考、将心比心啊。

你身为开发者要从原理的角度分析一个功能，建立它需要的复杂程度，评判一下它的合理性，再提问啊，因为这是个人开发者利用业余时间维护的一个很小的项目，**不是像powertoys那种微软有一个团伙在开发的公司项目，个人开发者不应该承受很多没有意义的多余工作。**

4. 你理解的对着呢，我不知道你是啥时候开始用的mouseinc，我大概是20年吧，那个时候还没有win11呐，**音量调节一直是调用的原生接口，而亮度调节从未调用过原生接口**，一直是这样以黑框和数值显示。

**区别就是，前者是不需要做ui的，后者需要自制ui**，这就是麻烦的所在。不明白的话，这也确实不太容易解释，你可以参考一下这个项目，[https://github.com/irzyxa/Volume2]()，这个软件有便携版，不向其他文件夹和注册表读写，我监视过，很干净，所以推荐用这个试一下就懂了，这个软件就实现了自定义亮度和音量的ui，你可以看一下自定义图形界面的ui是蛮复杂的一个新事情。

5. 话说回来吧，mouseinc最开始调节亮度是创建遮罩的形式，后来变成了wmi方式，**但是也从未直接调用过系统自带亮度条**，原因是啥，我没问过大佬，对此我有两种猜测吧，

    a. 系统亮度条从win10开始就从未有过显示数值的能力，猜大佬和我一样也嫌这玩意烦，就自制了简单可以显示数值的指示器。而音量本来就可以显示数值，就没动，直接调用系统原生。
    b. 可能是有什么技术上的困难，没法弄，不得已只好自己做了一个。

所以你看，就无论是a还是b，这都是一件很麻烦的事情

文末，致敬大佬吧，希望宽大处理

**Comment by SRH440** [comment]
我已经知道了 的确当一个伸手党并不好 我只是表达一下我的意愿罢了 但是在自己没有一点电脑能力的基础下说这个话的确不是很应该

感谢您的提示 我会尽量习惯这个黑框的

从 Windows 版邮件<https://go.microsoft.com/fwlink/?LinkId=550986>发送

发件人: ***@***.***>
发送时间: 2022年10月8日 13:18
收件人: ***@***.***>
抄送: ***@***.***>; ***@***.***>
主题: Re: [shuax/MouseInc.Settings] 可以将亮度调节的显示风格与win11 22H2版本的原生风格统一一下 (Discussion #448)


1.大佬已经退网了

2.加入图形界面需要设计和做进度条，可能需要集成一个或多个png，这是个全新的工作

3.你有这个想法可能是出于音量调节吧，那是系统自带的，不是大佬做的，这是不一样的

4.不显示数值有啥可好的，我十分感谢很多开发者做了第三方软件可以显示数值，没有数值连个常用参考都没有

―
